### PR TITLE
AV-1489: Docker & AWS CDK CI/CD pipelines

### PR DESCRIPTION
**Pull request by tonihW**
* update docker compose project to reflect changes in images and pipelines
* update aws cdk project to reflect changes in images and pipelines
* both compose and cdk refer to the file `docker/.env` for registry info and image tags
* delete cdk RegistryStack because now each docker image repo manages their respective registry
* use YAML anchors for shared env variables in docker compose project
* delete old circleci and github actions pipelines as unnecessary
* specify health checks for docker containers in compose project
* specify health checks for ecs tasks in cdk project
* finetune rolling deployment related configs in cdk project for smooth deploys

### CODE REVIEW

**Comment by tonihW** [comment]
* add CI/CD job for running E2E tests against local docker-compose env
* add CI/CD job for deploying AWS CDK stacks to development env

File: cloudformation/github-actions-stack.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        - PolicyName: s3-cdk-bucket-access
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action: "s3:*"
+                Resource: "arn:aws:s3:::cdktoolkit-stagingbucket-*"
+
+        - PolicyName: cfn-service-access
```
Comment (tonihW) [inline cloudformation/github-actions-stack.yml:78]: Is this OK? Or do we need more fine grained policies for the CDK deployment role?

The condition here makes sure that the administrator privileges are only given to the CloudFormation service.

**Comment by tonihW** [comment]
* deleted temporary dev-env CDK stack instances
* modified CDK stacks for installing them into existing environments, binds to existing resources
* added infratest-env CDK stack instances
* added infratest-env CD-pipeline

**Comment by tonihW** [comment]
* added tests for CDK project

File: .github/dependabot.yml [hunk 2]
```diff
@@ -23,11 +23,3 @@ updates:
...
     time: "03:00"
   open-pull-requests-limit: 10
-- package-ecosystem: npm
-  directory: "/modules/ytp-assets-common/"
-  schedule:
-    interval: daily
-    time: "03:00"
-  open-pull-requests-limit: 10
-  registries:
-  - npm-registry-npm-fontawesome-com
```
Comment (Zharktas) [inline .github/dependabot.yml:33]: Why this was removed ? It should be checking if there are updates to font awesome.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,112 +1,246 @@
...
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: setup docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: install nodejs v12
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (Zharktas) [inline .github/workflows/main.yml:34]: Could be newer node, v12 is in end of life next april.

File: .github/dependabot.yml (hunk 2 above)
Comment (tonihW) [reply .github/dependabot.yml:33]: @Zharktas I'm just going with the assumption that modules directory will be deleted from this repository when everything regarding this docker migration is done.

So shouldn't the dependabot check should be added to opendata-drupal and opendata-ckan repositories instead? Also I remember discussing with @bzar that later with the ckan 2.9 migration, we should probably have ytp-assets-common as a submodule.

**Comment by Zharktas** [comment]
Linting tasks in Github Actions should still be present. We still want to maintain code quality even though they are dockerized.

**Comment by tonihW** [comment]
> Linting tasks in Github Actions should still be present. We still want to maintain code quality even though they are dockerized.

Same as with the deleted dependabot check: shouldn't these linter jobs be transferred to the opendata-drupal and opendata-ckan repositories instead? That's why I deleted them from here.

File: .github/dependabot.yml (hunk 2 above)
Comment (Zharktas) [reply .github/dependabot.yml:33]: ah, true. Then this is not an issue.

**Review by Zharktas — APPROVED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (tonihW) [reply .github/workflows/main.yml:34]: Newer node won't work directly, I had it set to 16 before but the cypress project had some issues with it.

I'll leave it as a small issue for future, I'll fix it if I have time and remember.

**Pull request merged**